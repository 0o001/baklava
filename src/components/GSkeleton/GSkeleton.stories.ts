// @ts-nocheck
import { storiesOf } from '@storybook/vue';
import GSkeleton from './GSkeleton.vue';

storiesOf('GSkeleton', module)
  .addParameters({ component: GSkeleton })
  .lokiSkip('custom', () => ({
    components: { GSkeleton },
    template: `<div class="m-20" style="width: 350px">
      <GSkeleton circle width="60px" height="60px" />
      <div class="mt-12">
        <GSkeleton :count="5" />
      </div>
    </div>`,
  }))
  .add('static width/height (image)', () => ({
    components: { GSkeleton },
    template: `<div class="m-20" style="width: 350px">
      <GSkeleton width="180px" height="250px" />
    </div>`,
  }))
