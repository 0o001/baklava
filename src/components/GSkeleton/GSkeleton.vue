<template>
  <span>
    <span
      v-for="c in count"
      :key="c"
      class="g-skeleton"
      :style="styles()"
    />
  </span>
</template>

<script lang="ts">


export default {
  name: 'GSkeleton',
  props: {
    count: {
      default: 1,
      type: Number,
    },
    maxWidth: {
      default: 100,
      type: Number,
    },
    minWidth: {
      default: 80,
      type: Number,
    },
    width: {
      default: null,
      type: String,
    },
    height: {
      default: '1em',
      type: String,
    },
    circle: {
      default: false,
      type: Boolean,
    },
  },
  methods: {
    styles () {
      return {
        height: this.height,
        width: this.width || this.customWidth(),
        'border-radius': this.circle ? '50%' : '0%',
      };
    },
    customWidth () {
      return `${Math.floor((Math.random() * (this.maxWidth - this.minWidth)) + this.minWidth)}%`;
    },
  },
};
</script>

<style lang="scss" scoped>
.g-skeleton {
  display: inline-block;
  position: relative;
  overflow: hidden;
  vertical-align: middle;
  background-color: var(--bg-grey-200);

    &::after {
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      transform: translateX(-100%);
      background-image: linear-gradient(
        90deg,
        rgba(#fff, 0) 0,
        rgba(#fff, 0.2) 20%,
        rgba(#fff, 0.5) 60%,
        rgba(#fff, 0)
      );
      animation: shimmer 2s infinite;
      content: '';
    }

  @keyframes shimmer {
    100% {
      transform: translateX(100%);
    }
  }
}
</style>
