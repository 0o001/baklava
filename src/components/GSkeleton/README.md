# GSkeleton

## Props

| Name        | Type      | Description | Default |
| ----------- | --------- | ----------- | ------- |
| `count`     | `Number`  |             | `1`     |
| `max-width` | `Number`  |             | `100`   |
| `min-width` | `Number`  |             | `80`    |
| `width`     | `String`  |             | `null`  |
| `height`    | `String`  |             | `"1em"` |
| `circle`    | `Boolean` |             | `false` |

## Methods

### styles()

**Syntax**

```typescript
styles(): unknown
```

### customWidth()

**Syntax**

```typescript
customWidth(): unknown
```

