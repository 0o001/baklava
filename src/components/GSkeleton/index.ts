import GSkeleton from './GSkeleton.vue';

export default GSkeleton;
