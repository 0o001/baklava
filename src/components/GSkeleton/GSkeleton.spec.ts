import { mount } from '@vue/test-utils';
import GSkeleton from "./";

describe('GSkeleton', () => {
  let wrapper: any;
  let mathCopy: any;

  mathCopy = Object.create(global.Math);
  mathCopy.random = () => 0.5;
  global.Math = mathCopy;

  beforeEach(() => {
    wrapper = mount(GSkeleton);
  });

  it('should match snapshot with default style', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render correctly simple skeleton', () => {
    const skeletonStyles = wrapper.find('.g-skeleton').attributes('style');

    expect(skeletonStyles).toEqual('height: 1em; width: 90%; border-radius: 0%;');
  });

  it('should render with given count', async () => {
    wrapper.setProps({
      count: 5,
    });

    await wrapper.vm.$nextTick();

    const skeletonCount = wrapper.findAll('.g-skeleton').length;

    expect(skeletonCount).toEqual(5);
  });

  it('should render with given width and height', async () => {
    wrapper.setProps({
      width: '50px',
      height: '100px',
    });

    await wrapper.vm.$nextTick();

    const skeletonStyles = wrapper.find('.g-skeleton').attributes('style');

    expect(skeletonStyles).toEqual('height: 100px; width: 50px; border-radius: 0%;');
  });

  it('should set border radius if circle props is true', async () => {
    wrapper.setProps({
      circle: true,
    });

    await wrapper.vm.$nextTick();

    const skeletonStyles = wrapper.find('.g-skeleton').attributes('style');

    expect(skeletonStyles).toEqual('height: 1em; width: 90%; border-radius: 50%;');
  });
})
