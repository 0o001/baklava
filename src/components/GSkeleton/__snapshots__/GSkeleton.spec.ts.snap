// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GSkeleton should match snapshot with default style 1`] = `
<span>
  <span
    class="g-skeleton"
    style="height: 1em; width: 90%; border-radius: 0%;"
  />
</span>
`;
